### DISCUSSION TITLE
Composer warning

### DISCUSSION THREAD

**Discussion by karaka200** [symfony/symfony#44793] [Q&A]
Hi. When I use any command from the Composer (like `composer update`) , this warning display, what's wrong?

```
composer/package-versions-deprecated contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
Do you trust "composer/package-versions-deprecated" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?]
```

I see `https://getcomposer.org/allow-plugins` but I did not notice anything.

**Comment by karaka200** [comment]
> Hi. When I use any command from the Composer (like `composer update`) , this warning display, what's wrong?
> 
> ```
> composer/package-versions-deprecated contains a Composer plugin which is currently not in your allow-plugins config. See https://getcomposer.org/allow-plugins
> Do you trust "composer/package-versions-deprecated" to execute code and wish to enable it now? (writes "allow-plugins" to composer.json) [y,n,d,?]
> ```
> 
> I see `https://getcomposer.org/allow-plugins` but I did not notice anything.

When I write `yes`, a part is added to `allow-plugins`. Doesn't it pose a security problem?

```
    "config": {
        "allow-plugins": {
            "symfony/flex": true,
            "symfony/runtime": true,
            "composer/package-versions-deprecated": true
        },
```

**Reply by ruudk** [reply]
This is because you use Composer 2.2.0RC (preview). It's a new feature they added there. I've submitted PR's to various repos to add this `allow-plugins` entries. You could do the same. 

Keep in mind that Symfony will drop this dependency in the future, see https://github.com/symfony/symfony/issues/44726

**Reply by aijcoa** [reply]
Does adding ` "composer/package-versions-deprecated": true` pose a secuirty problem?

**Reply by ruudk** [reply]
No, not for that package. But it depends per package, that's why it was added.
